evm/v0.5/testv2/Coordinator.test.ts

```diff
@@ -0,0 +1,909 @@
+      getterSetterFactory,
+      'requestedBytes32',
+    )
+
+    describe('cooperative consumer', () => {
+      const message = h.toBytes32String('Hello World!')
+      beforeEach(async () => {
+        mock = await getterSetterFactory.connect(roles.defaultAccount).deploy()
+        const payload = h.executeServiceAgreementBytes(
+          sAID,
+          mock.address,
+          fHash,
+          1,
+        )
+        const tx = await link.transferAndCall(
+          coordinator.address,
+          agreement.payment,
+          payload,
+          { value: 0 },
+        )
+        const log = await h.getLog(tx, 2)
+        request = h.decodeRunRequest(log)
+      })
+
+      describe('when called by a non-owner', () => {
+        // Turn this test on when multiple-oracle response aggregation is enabled
+        xit('raises an error', async () => {
+          coordinator = coordinator.connect(roles.stranger)
+          await h.assertActionThrows(
+            coordinator.fulfillOracleRequest(request.id, message),
+          )
+        })
+      })
+
+      describe('when called by an owner', () => {
+        beforeEach(() => {
+          coordinator = coordinator.connect(roles.oracleNode)
+        })
+
+        it('raises an error if the request ID does not exist', async () => {
+          const invalidRequestId = h.toBytes32String('deadbeef')
+          await h.assertActionThrows(
+            coordinator.fulfillOracleRequest(invalidRequestId, message),
+          )
+        })
+
+        it('sets the value on the requested contract', async () => {
+          await coordinator.fulfillOracleRequest(request.id, message)
+          const mockRequestId = await mock.requestId()
+          assert.equal(h.toHex(request.id), mockRequestId)
+          const currentValue = await mock.getBytes32()
+          assert.equal('Hello World!', h.parseBytes32String(currentValue))
+        })
+
+        it('reports errors from the aggregator, such as double-reporting', async () => {
+          const firstMessage = h.toBytes32String('First message!')
+          const seccondMessage = h.toBytes32String('Second message!!')
+          await coordinator.fulfillOracleRequest(request.id, firstMessage)
+          await h.assertActionThrows(
+            coordinator.fulfillOracleRequest(request.id, seccondMessage),
+            'oracle already reported',
+          )
+        })
+      })
+    })
+
+    describe('with a malicious requester', () => {
+      let mock: Instance<MaliciousRequesterFactory>
+      const paymentAmount = h.toWei('1')
+
+      beforeEach(async () => {
+        mock = await maliciousRequesterFactory
+          .connect(roles.defaultAccount)
+          .deploy(link.address, coordinator.address)
+        await link.transfer(mock.address, paymentAmount)
+      })
+
+      xit('cannot cancel before the expiration', async () => {
+        await h.assertActionThrows(
+          mock.maliciousRequestCancel(sAID, 'doesNothing(bytes32,bytes32)'),
```

Comment:
lets use `mock.interface.functions.doesNothing.signature` here to avoid hard coding